types/type_map.go

```diff
@@ -99,3 +103,16 @@ func (t typeMap) Validate() error {
 	}
 	return t.value.Validate()
 }
+
+func genName(baseName string, key bool) string {
+	postfix := "value"
+	if key {
+		postfix = "key"
+	}
+	digest := md5.New()
+	n, err := digest.Write([]byte(baseName + postfix))
```

Comment:
1 and 2. you just count bytes. What uniqueness you guarantee?
2. This global var exists only on generation step. Generation is a sequential process. Why not?